### PR TITLE
SIMPLY-3696: New APIs for Library Finder

### PR DESCRIPTION
**Pull request by io7m**
**What's this do?**
This replaces the existing library registry parser with a real OPDS
2.0 parser. This also necessitated upgrading to the latest AudioBook
API to avoid a dependency conflict. It then adds a basic API for querying
the library registry. 

Currently, the API merely calls the existing `/libraries` endpoint, and annotates
each entry with fake location and distance data. This will be replaced
with a call to a different endpoint when the server work is completed.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SMA-7

**How should this be tested? / Do these changes have associated tests?**
Test that the library registry currently behaves identically to the way it used to behave.

**Dependencies for merging? Releasing to production?**
None.

**Has the application documentation been updated for these changes?**
The module list was updated.

**Did someone actually run this code to verify it works?**
@io7m Has been trying it.

### CODE REVIEW

File: simplified-opds2-r2/src/main/java/org/nypl/simplified/opds2/r2/internal/OPDS2ParserR2.kt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  override fun parse(): ParseResult<OPDS2Feed> {
+    return try {
+      val data = OPDS2Parser.parse(stream.readBytes(), uri.toURL())
+      ParseResult.Failure(
+        warnings = listOf(),
+        errors = listOf(
+          ParseError(
+            source = this.uri,
+            message = "Not implemented yet!",
+            line = 0,
```
Comment (io7m) [inline simplified-opds2-r2/src/main/java/org/nypl/simplified/opds2/r2/internal/OPDS2ParserR2.kt:26]: I've left this parser in place in order to keep our options open with regards to OPDS 2.0 parsers. I'd like to finish this at a slightly later date and do a competitive analysis between the R2 and Irradia parsers. Right now, we can't use R2 for the accounts registry as it doesn't support the custom roles and other things the accounts registry feeds use.

File: simplified-tests/src/test/java/org/nypl/simplified/tests/opds2/OPDS2ParserContract.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.nypl.simplified.opds2.parser.api.OPDS2ParsersType
+import org.nypl.simplified.parser.api.ParseResult
+import org.slf4j.Logger
+import java.io.FileNotFoundException
+import java.io.InputStream
+import java.net.URI
+
+abstract class OPDS2ParserContract {
```
Comment (qnga) [inline simplified-tests/src/test/java/org/nypl/simplified/tests/opds2/OPDS2ParserContract.kt:13]: This is interesting. I used to use parametrized tests in such cases, is there any specific rationale behind your choice?

**Review by qnga — APPROVED** [review]
Looks fine!

File: simplified-tests/src/test/java/org/nypl/simplified/tests/opds2/OPDS2ParserContract.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (io7m) [reply simplified-tests/src/test/java/org/nypl/simplified/tests/opds2/OPDS2ParserContract.kt:13]: Hah, didn't think of that! We only just moved from junit 4 to junit 5, and I guess I was still thinking in terms of junit 4 (without extensions)... JUnit 5 has an API in the core for injecting different values into tests.

**Pull request merged**